### DISCUSSION TITLE
Outlook OWA time bug on mobile

### DISCUSSION THREAD

**Discussion by jekuer** [add2cal/add-to-calendar-button#113] [Ideas / Requests]
On mobile devices (or simply small screens), the Outlook web app runs into a bug, where time information gets messed up.
This is not on our, but Microsoft's side.

As for now, we decided to go for a workaround, where the button would re-route for the iCal version on mobile devices, when aiming for Microsoft services. This is not ideal, but at the moment the only proper way, which can run on the client side.

This discussion is opened to keep track of this issue and maybe someone comes up with a better workaround solution.

Best,
Jens

--- 

You could test the issue with the following URL - calling it on a mobile device. 
No matter the timezone, it should still show a starting time at minute 15, but somehow always rounds it randomly.
[https://outlook.live.com/calendar/0/deeplink/compose?path=%2Fcalendar%2Faction%2Fcompose&rru=addevent&startdt=2023-02-14T09%3A15%3A00Z&enddt=2023-02-18T22%3A30%3A00Z&subject=Test&location=World%20Wide%20Web&body=Testing](https://outlook.live.com/calendar/0/deeplink/compose?path=%2Fcalendar%2Faction%2Fcompose&rru=addevent&startdt=2023-02-14T09%3A15%3A00Z&enddt=2023-02-18T22%3A30%3A00Z&subject=Test&location=World%20Wide%20Web&body=Testing)

**Comment by jekuer** [comment]
It feels like this is connected to the time zone. If we would know how to provide this, it might solve the issue. Unfortunately, this is quite some mystery.

**Comment by sc-kknowles** [comment]
@jekuer there is a bug with this workaround. On iOS mobile the type now gets set to 'ical' but ical will not be in the atcbStates identifier array. 
![image](https://user-images.githubusercontent.com/98483359/224129233-867dbb3d-4688-41e2-afbf-fba628b496be.png).

**Reply by jekuer** [reply]
Thanks a lot!
Should be fixed with v2.2.4

**Reply by MiguelBrito1086** [reply]
But this bug is an outlook thing right? Mobile browser/view literally strips the url and doesn't acknowledge start time. Date is there. End date and time work... Just the start time is unacknowledged.

**Reply by jekuer** [reply]
Correct. Nothing we can do about the overall issue with mobile Outlook messing around with the data.
In case you know somebody at Microsoft, please push 🙏🏼 .

I already went through multiple "support" cycles - ending up at some technical support expert telling me to contact the person I spoke to first, which guided me to place B, which brought me to X, which led me to this expert. 😕 
I guess there is no real issue tracking for Microsoft products outside of Azure.

**Comment by jekuer** [comment]
This seems to be fixed now.
Therefore, rolling back the workaround with latest patch release.
